src/common/protobuf_utils.cpp

```diff
@@ -77,6 +77,56 @@ namespace mesos {
 namespace internal {
 namespace protobuf {
 
+
+UnionValidator::UnionValidator(
+  const google::protobuf::Descriptor* descriptor,
+  const google::protobuf::EnumDescriptor* typeDescriptor)
+{
+  typeDescriptor_ = typeDescriptor;
+  const auto* unknownTypeValueDescriptor = typeDescriptor->FindValueByNumber(0);
```

Comment:
Consider leaving a newline after `typeDescriptor_ = typeDescriptor;` and inserting a short comment after this section, i.e.:
```
// If present, the the `0` value of the `Type` enum must be named "UNKNOWN".
const auto* unknownTypeValueDescriptor = typeDescriptor->FindValueByNumber(0);
...
```

Reply:
Now it should be much more clear from the code of the UnionValidator constructor to which field the typeDescriptor corresponds.